lib/sysloggable/logger.rb

```diff
@@ -0,0 +1,66 @@
+module Sysloggable
+  class Logger
+    SEVERITIES = {
+      debug: ::Logger::DEBUG,
+      info: ::Logger::INFO,
+      warn: ::Logger::WARN,
+      error: ::Logger::ERROR,
+      unknown: ::Logger::UNKNOWN
+    }.freeze
+
+    def initialize(options)
+      @options = options
+    end
+
+    [:debug, :info, :warn, :error, :fatal, :unknown].each do |logger_method|
+      define_method logger_method do |*args, &block|
+        add(SEVERITIES[logger_method], *args, &block)
+      end
+    end
+
+    private
+
+    def logger
+      return @logger if defined?(@logger)
+
+      @logger = Container['lib.syslogger'].new(
+        @options.fetch(:ident),
+        ::Syslog::LOG_PID | ::Syslog::LOG_CONS,
+        ::Syslog::LOG_LOCAL0
+      )
+
+      @logger.level = @options.fetch(:level, ::Logger::INFO)
+      @logger.push_tags(@options[:tags]) if @options[:tags]
+
+      @logger
+    end
+
+    def add(severity, message, params = {})
+      if block_given?
+        beginning = Time.now
+        yield params
+        duration = Time.now - beginning
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.